flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/strategy/TestingSchedulingExecutionVertex.java

```diff
@@ -94,6 +94,9 @@ void addConsumedPartition(TestingSchedulingResultPartition consumedPartition) {
                 ConsumedPartitionGroup.fromSinglePartition(consumedPartition.getId());
 
         consumedPartition.registerConsumedPartitionGroup(consumedPartitionGroup);
+        if (consumedPartition.getState() == ResultPartitionState.CONSUMABLE) {
```

Comment:
can we put this condition block into `consumedPartition.registerConsumedPartitionGroup`?